Composer/packages/client/src/components/Auth/AuthCard.tsx

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import formatMessage from 'format-message';
+import { Fragment, useEffect, useState } from 'react';
+import { ActionButton } from 'office-ui-fabric-react/lib/Button';
+import { Callout } from 'office-ui-fabric-react/lib/Callout';
+import { Persona, PersonaSize } from 'office-ui-fabric-react/lib/Persona';
+import { ILinkStyles, Link } from 'office-ui-fabric-react/lib/Link';
+import { useRecoilValue } from 'recoil';
+import { NeutralColors } from '@uifabric/fluent-theme';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+import { FontSizes } from 'office-ui-fabric-react/lib/Styling';
+
+import {
+  primaryTokenState,
+  dispatcherState,
+  showAuthDialogState,
+  showTenantDialogState,
+} from '../../recoilModel/atoms';
+import { isTokenExpired, decodeToken, setTenantId } from '../../utils/auth';
+import { zIndices } from '../../utils/zIndices';
+
+import { AuthDialog } from './AuthDialog';
+import { TenantDialog } from './TenantDialog';
+
+const styles = {
+  username: {
+    display: 'block',
+    fontWeight: 'bold',
+  },
+  email: {
+    display: 'block',
+    fontSize: FontSizes.small,
+  },
+  link: {
+    root: {
+      display: 'block',
+      fontSize: FontSizes.small,
+      marginTop: 5,
+    },
+  },
+  logoutLink: {
+    root: {
+      display: 'block',
+      fontSize: FontSizes.small,
+    },
+  },
+};
+
+export const AuthCard: React.FC = () => {
+  const token = useRecoilValue(primaryTokenState);
+  const [authCardVisible, setAuthCardVisible] = useState<boolean>(false);
+  const { refreshLoginStatus, requireUserLogin, logoutUser, setShowAuthDialog, setShowTenantDialog } = useRecoilValue(
+    dispatcherState
+  );
+  const showAuthDialog = useRecoilValue(showAuthDialogState);
+  const showTenantDialog = useRecoilValue(showTenantDialogState);
+  const [currentUser, setCurrentUser] = useState<any>(undefined);
```

Comment:
Why wouldn't this also live in the recoil state?

Reply:
the current user? I agree! I am going to move it there.

Reply:
done! i also refactored the state a bit so that we have nice accessors like `currentUser` and `isAuthenticated` that are also available via the extension client.